tests/envs/register_during_make_env.py

```diff
@@ -0,0 +1,22 @@
+import gym
+
+
+class RegisterDuringMakeEnv(gym.Env):
+    """Used in `test_registration.py` to check if `env.make` can import and register an env"""
+
+    def __init__(self):
+        super().__init__()
+        self.act_space = gym.spaces.Discrete(2)
+        self.obs_space = gym.spaces.Discrete(2)
```

Comment:
Why not `self.action_space` and `self.observation_space`?

Reply:
Hmm just instinctively did it, I've seen a trend of using shorthand type names when the variables don't really matter. Like, `o, r, d, i = env.step(action)` in tests when we're not really going to care about the variables themselves and writing the whole thing can get verbose. But in hindsight having proper names keeps things uniform, I will modify it.

Reply:
Since it's part of the official API, it really should be `self.action_space`. I think there is currently a PR that would raise an exception if the environment is instantiated via `make` without `self.action_space`.